spanner/api/Spanner/CancelBackupOperation.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright 2020 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START spanner_cancel_backup_create]
+using Google.Cloud.Spanner.Admin.Database.V1;
+using Google.Cloud.Spanner.Common.V1;
+using Google.LongRunning;
+using Google.Protobuf.WellKnownTypes;
+using System;
+
+namespace GoogleCloudSamples.Spanner
+{
+    public class CancelBackupOperation
+    {
+        public static object SpannerCancelBackupOperation(
+            string projectId, string instanceId, string databaseId, string backupId)
+        {
+            // Create the DatabaseAdminClient instance.
+            DatabaseAdminClient databaseAdminClient = DatabaseAdminClient.Create();
+
+            // Initialize request parameters.
+            Backup backup = new Backup
+            {
+                Database = DatabaseName.Format(projectId, instanceId, databaseId),
+                ExpireTime = DateTime.UtcNow.AddDays(14).ToTimestamp()
+            };
+            InstanceName parentAsInstanceName = InstanceName.FromProjectInstance(projectId, instanceId);
+
+            // Make the CreateBackup request.
+            Operation<Backup, CreateBackupMetadata> response =
+                databaseAdminClient.CreateBackup(parentAsInstanceName, backup, backupId);
+
+            // Create the OperationsClient instance and execute CancelOperation.
+            OperationsClient operationsClient = OperationsClient.Create();
+            operationsClient.CancelOperation(response.Name);
```

Comment:
@laljikanjareeya I added a test for this sample but I'm having trouble getting it to work. This call returns a "CANCELLED" error and the sample exits because we don't have the call around a try/catch. But it doesn't make sense to me why the cancel operation is cancelled. If you could run the test and take a look, that would be great.

Reply:
@skuruppu I am also getting below exception:
`Status(StatusCode=Cancelled, Detail="Received http2 header with status: 404")`.
Also tried with `CancelOperationRequest` but same error, digging more into it will let you know further findings.

Reply:
I managed to find a fix. I don't know why but the error is 404 so for some reason the backend is not able to find the operation by name. But you can also cancel an operation by just doing Operation.Cancel() so I changed it to that and everything works. It's also much simpler.

Reply:
Ok so the real problem here is that you shouldn't be using the OperationsClient directly. I'm not on a computer but I'll look later and try to see exactly how this should be done. For the time being see a response to a similar issue here. Sorry for not realizing earlier.
 https://github.com/googleapis/google-cloud-dotnet/issues/4843#issuecomment-611731182

Reply:
Oh, I think Shanika's comment and mine got crossed. Shanika, the solution that you found is the right one.
Using the OperationsClient directly would give a 404 because that's an OperationsClient that we use for all long running operations from all APIs, so as you were using it it doesn't know it needs to talk to Spanner.
What you did is the cleanest solution, but if you want to use an OperationsClient for dealing with Backup creation operations the you need to use the one in `databaseAdminClient.CreateBackupOperationsClient`.

Reply:
Ah ok that makes a lot of sense now. Thanks for the explanation. I don't think we every want our users to use the OperationsClient directly. So having the samples doing the simplest thing, even for creating a backup, is the way to go.